src/ashley/machina_extensions/forum/views.py

```diff
@@ -15,20 +17,132 @@
+        """Gets order requested in url"""
+        ordering = self.request.GET.get(
+            ORDER_VAR, self.get_default_index_order().get("index")
+        )
+        try:
+            _none, pfx, idx = ordering.rpartition("-")
+            # returns default column if out of index order
+            if (int)(idx) >= len(self.list_display):
+                idx = self.get_default_index_order().get("col")
+            return pfx, (int)(idx)
+        except ValueError:
+            # param exists but value is unexpected, use default order
+            _none, pfx, idx = (
+                self.get_default_index_order().get("index").rpartition("-")
+            )
+            return pfx, (int)(idx)
+
+    def get_queryset(self):
+        """Returns the list of items for this view ordered by asked param."""
+        query = super().get_queryset()
+        return query.order_by(self.get_ordering())
+
+    def get_context_data(self, **kwargs):
+        """Returns the context data to provide to the template."""
+        context = super().get_context_data(**kwargs)
+        # add sort header links containing asc, desc orders
+        context["header"] = list(self.result_headers())
+        # used for pagination to keep the order
+        if self.request.GET.get(ORDER_VAR):
+            context["order"] = f"{ORDER_VAR}={self.request.GET.get(ORDER_VAR)}"
+        return context
+
+    def result_headers(self):
+        """Builds each link and css class attribute for the header depending on current order"""
+        pfx, idx = self.get_default_ordering()
+
+        for index, _field_name in enumerate(self.list_display):
+            order_type = ""
+            new_order_type = "asc"
+            div_classes = ""
+            query_string = self.get_query_string({ORDER_VAR: "-" + str(index)})
+            # Is it currently being sorted on?
+            is_sorted = index == idx
+            if is_sorted:
+                order_type = "desc" if pfx == "-" else "asc"
+                div_classes = f"sorted {order_type}ending"
+                new_order_type = {"asc": "desc", "desc": "asc"}[order_type]
+                query_string = self.get_query_string(
+                    {ORDER_VAR: ("-" if pfx == "" else "") + str(index)}
+                )
+
+            yield {
+                "sorted": is_sorted,
+                "ascending": new_order_type == "desc",
+                "url_order": query_string,
+                "class_attrib": div_classes,
+            }
+
+    def get_query_string(self, new_params):
+        """
+        Build the query_string for this link keeping all the query params and replacing
+        or adding new_params
+        """
+        param = self.params.copy()
+        for key, value in new_params.items():
+            if value is None:
+                if key in param:
+                    del param[key]
+            else:
+                param[key] = value
+
+        return "?%s" % urlencode(sorted(param.items()))
```

Comment:
Unless I am mistaken, I don't see any case in which these lines are executed : 
```python
            if value is None:
                if key in param:
                    del param[key]
```
If my assumption is correct, I suggest the following rewriting of this function:

```suggestion
        param = {**self.params, **new_params}
        return "?%s" % urlencode(sorted(param.items()))
```

Reply:
I wish I had done this one, nice 👍